UI/ChatUI/ChatUI/ChatUI/MainPage.xaml

```diff
@@ -0,0 +1,427 @@
+												<win:FadeOutThemeAnimation TargetName="ContentBorder" />
+											</Storyboard>
+										</VisualState>
+										<VisualStateGroup.Transitions>
+											<VisualTransition GeneratedDuration="0:0:0.2" To="NotDragging" />
+										</VisualStateGroup.Transitions>
+									</VisualStateGroup>
+								</VisualStateManager.VisualStateGroups>
+							</Grid>
+						</ControlTemplate>
+					</Setter.Value>
+				</Setter>
+			</Style>
+
+			<local:MessageTemplateSelector x:Key="MessageTemplateSelector">
+				<local:MessageTemplateSelector.MyMessageTemplate>
+					<DataTemplate>
+						<StackPanel Margin="20,15,0,0">
+							<Border
+								MinWidth="300"
+								MinHeight="100"
+								Padding="10"
+								Background="DeepPink"
+								CornerRadius="10">
+								<Grid>
+									<Grid.RowDefinitions>
+										<RowDefinition Height="*" />
+										<RowDefinition Height="*" />
+										<RowDefinition Height="Auto" />
+									</Grid.RowDefinitions>
+									<TextBlock FontWeight="Bold" Text="{Binding ContactName}" />
+									<TextBlock
+										Grid.Row="1"
+										Text="{Binding Text}"
+										TextWrapping="WrapWholeWords" />
+									<TextBlock
+										Grid.Row="2"
+										HorizontalAlignment="Right"
+										Text="{Binding UserFriendlyTimestamp}" />
+								</Grid>
+							</Border>
+							<Polygon
+								Margin="0,0,10,0"
+								HorizontalAlignment="Right"
+								Fill="DeepPink"
+								Points="0,0 10,0 10,10"
+								Visibility="{Binding IsLastInSequence, Converter={StaticResource BooleanToVisibility}}" />
+						</StackPanel>
+					</DataTemplate>
+				</local:MessageTemplateSelector.MyMessageTemplate>
+				<local:MessageTemplateSelector.OtherMessageTemplate>
+					<DataTemplate>
+						<StackPanel Margin="0,15,20,0">
+							<Border
+								MinWidth="300"
+								MinHeight="100"
+								Padding="10"
+								Background="DeepSkyBlue"
+								CornerRadius="10">
+								<Grid>
+									<Grid.RowDefinitions>
+										<RowDefinition Height="*" />
+										<RowDefinition Height="*" />
+										<RowDefinition Height="Auto" />
+									</Grid.RowDefinitions>
+									<TextBlock FontWeight="Bold" Text="{Binding ContactName}" />
+									<TextBlock
+										Grid.Row="1"
+										Text="{Binding Text}"
+										TextWrapping="WrapWholeWords" />
+									<TextBlock
+										Grid.Row="2"
+										HorizontalAlignment="Right"
+										Text="{Binding UserFriendlyTimestamp}" />
+								</Grid>
+							</Border>
+							<Polygon
+								Margin="10,0,0,0"
+								HorizontalAlignment="Left"
+								Fill="DeepSkyBlue"
+								Points="0,0 10,0 0,10"
+								Visibility="{Binding IsLastInSequence, Converter={StaticResource BooleanToVisibility}}" />
+						</StackPanel>
+					</DataTemplate>
+				</local:MessageTemplateSelector.OtherMessageTemplate>
+			</local:MessageTemplateSelector>
+		</ResourceDictionary>
+	</Page.Resources>
+	
+	<ScrollViewer x:Name="scrollViewer"
```

Comment:
you are using PascalCase everywhere else. just stick to it?
you can rename it to RootScrollViewer to avoid name collision with its class